nwss_wastewater/delphi_nwss/run.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+"""Functions to call when running the function.
+
+This module should contain a function called `run_module`, that is executed
+when the module is run with `python -m MODULE_NAME`.  `run_module`'s lone argument should be a
+nested dictionary of parameters loaded from the params.json file.  We expect the `params` to have
+the following structure:
+    - "common":
+        - "export_dir": str, directory to write daily output
+        - "log_filename": (optional) str, path to log file
+        - "log_exceptions" (optional): bool, whether to log exceptions to file
+    - "indicator": (optional)
+        - "wip_signal": (optional) Any[str, bool], list of signals that are
+            works in progress, or True if all signals in the registry are works
+            in progress, or False if only unpublished signals are.  See
+            `delphi_utils.add_prefix()`
+        - "test_file" (optional): str, name of file from which to read test data
+        - "token": str, authentication for upstream data pull
+    - "archive" (optional): if provided, output will be archived with S3
+        - "aws_credentials": Dict[str, str], AWS login credentials (see S3 documentation)
+        - "bucket_name: str, name of S3 bucket to read/write
+        - "cache_dir": str, directory of locally cached data
+"""
+import time
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+from delphi_utils import S3ArchiveDiffer, get_structured_logger, create_export_csv
+from delphi_utils.nancodes import add_default_nancodes
+
+from .constants import GEOS, SIGNALS
+from .pull import pull_nwss_data
+
+
+def sum_all_nan(x):
+    """Return a normal sum unless everything is NaN, then return that."""
+    sum_of = np.nansum(x)
+    all_nan = np.isnan(x).all()
+    if all_nan:
+        return np.nan
+    return sum_of
```

Comment:
dont compute the sum unless you have to
```suggestion
    if np.isnan(x).all():
        return np.nan
    return np.nansum(x)
```